internal/scope/machine.go

```diff
@@ -282,6 +282,38 @@ func (m *MachineScope) GetSSHPublicKeys() []infrav1.SSHPublicKey {
 	return nil
 }
 
+// GetBasicAuthToken will fetch the BasicAuthSecret on the MvmCluster and
+// and return the token for the given host.
+// If no secret or no value is found, an empty string is returned.
+func (m *MachineScope) GetBasicAuthToken(addr string) (string, error) {
+	placement := m.MvmCluster.Spec.Placement
+	if placement.StaticPool == nil || placement.StaticPool.BasicAuthSecret == "" {
+		return "", nil
+	}
+
+	tokenSecret := &corev1.Secret{}
+	key := types.NamespacedName{
+		Name:      m.MvmCluster.Spec.Placement.StaticPool.BasicAuthSecret,
+		Namespace: m.MvmCluster.Namespace,
+	}
+
+	if err := m.client.Get(m.ctx, key, tokenSecret); err != nil {
+		return "", err
+	}
+
+	host := strings.Split(addr, ":")[0]
+	// If it's not there, that's fine; we will log and return an empty string
+	token := string(tokenSecret.Data[host])
+
+	if token == "" {
+		m.V(2).Info( //nolint:gomnd // this magic number is fine
```

Comment:
I'd actually take off the verbosity of 2.

```suggestion
		m.Info( 
```

Reply:
eh i'll do that in the next one 😈